test/spmd/tensor/test_matrix_ops.py

```diff
@@ -223,19 +233,28 @@ def test_placement_comb(
         shard0_spec = Shard(0)
         shard1_spec = Shard(1)
         shard2_spec = Shard(2)
-        shard_specs = [shard0_spec, shard1_spec, shard2_spec]
-        shard_specs_comb = list(itertools.product(shard_specs, shard_specs))
+        replica_spec = Replicate()
+        placement_specs = [shard0_spec, shard1_spec, shard2_spec, replica_spec]
+        shard_specs_comb = list(
+            itertools.product(placement_specs, placement_specs)
+        )
         passlist = [
             (shard0_spec, shard0_spec),
+            (shard0_spec, replica_spec),
+            (shard1_spec, replica_spec),
+            (shard2_spec, replica_spec),
+            (replica_spec, shard0_spec),
+            (replica_spec, shard1_spec),
+            (replica_spec, shard2_spec),
             (shard2_spec, shard1_spec),
+            (replica_spec, replica_spec),
```

Comment:
Can we also add backward in the test like what we have in https://github.com/pytorch/tau/pull/382/files#diff-156f4bb8984b2bbd435302d728ba6a9607d2a197873c03dd0c8a73dff5e8a06aR94?

Reply:
Sure, will add backward tests in a follow up PR